cdap-api/src/main/java/co/cask/cdap/api/dataset/DatasetSpecification.java

```diff
@@ -194,21 +208,25 @@ public DatasetSpecification setOriginalProperties(DatasetProperties originalProp
    * @return a new spec that is the same as this one, with the original properties set to the provided properties.
    */
   public DatasetSpecification setOriginalProperties(Map<String, String> originalProps) {
-    return new DatasetSpecification(name, type, originalProps, properties, datasetSpecs);
+    return new DatasetSpecification(name, type, description, originalProps, properties, datasetSpecs);
+  }
+
+  public DatasetSpecification setDescription(String description) {
```

Comment:
Description is per dataset instance, and configure method is per dataset type, right? 

Description will be provided while creating a dataset instance. I think this description should override whatever description the type would have set. Hence I'm setting it after the configure method has returned the spec.